README.txt

Please refer to manual you find in doc/manual.sxw


ChangeLog
2010-08-13 Peter Proell  <peter@alinbu.net>

	* initial code generated with kickstarter


ext_tables.php
<?php
if (!defined ('TYPO3_MODE')) {
	die ('Access denied.');
}

t3lib_div::loadTCA('tt_content');
$TCA['tt_content']['types']['list']['subtypes_excludelist'][$_EXTKEY.'_pi1']='layout,select_key';


t3lib_extMgm::addPlugin(array(
	'LLL:EXT:tweetbutton/locallang_db.xml:tt_content.list_type_pi1',
	$_EXTKEY . '_pi1',
	t3lib_extMgm::extRelPath($_EXTKEY) . 'ext_icon.gif'
),'list_type');


if (TYPO3_MODE == 'BE') {
	$TBE_MODULES_EXT['xMOD_db_new_content_el']['addElClasses']['tx_tweetbutton_pi1_wizicon'] = t3lib_extMgm::extPath($_EXTKEY).'pi1/class.tx_tweetbutton_pi1_wizicon.php';
}

t3lib_extMgm::addStaticFile($_EXTKEY,'static/tweetbutton/', 'tweetbutton');

// Activating flexforms
$TCA['tt_content']['types']['list']['subtypes_addlist'][$_EXTKEY . '_pi1'] = 'pi_flexform';
t3lib_extMgm::addPiFlexFormValue($_EXTKEY . '_pi1', 'FILE:EXT:' . $_EXTKEY . '/pi1/flexform_ds.xml');
?>

pi1/class.tx_tweetbutton_pi1.php
<?php
/***************************************************************
*  Copyright notice
*
*  (c) 2010 Peter Proell <peter@alinbu.net>
*  All rights reserved
*
*  This script is part of the TYPO3 project. The TYPO3 project is
*  free software; you can redistribute it and/or modify
*  it under the terms of the GNU General Public License as published by
*  the Free Software Foundation; either version 2 of the License, or
*  (at your option) any later version.
*
*  The GNU General Public License can be found at
*  http://www.gnu.org/copyleft/gpl.html.
*
*  This script is distributed in the hope that it will be useful,
*  but WITHOUT ANY WARRANTY; without even the implied warranty of
*  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
*  GNU General Public License for more details.
*
*  This copyright notice MUST APPEAR in all copies of the script!
***************************************************************/
/**
 * [CLASS/FUNCTION INDEX of SCRIPT]
 *
 * Hint: use extdeveval to insert/update function index above.
 */

require_once(PATH_tslib.'class.tslib_pibase.php');


/**
 * Plugin 'Tweet button' for the 'tweetbutton' extension.
 *
 * @author	Peter Proell <peter@alinbu.net>
 * @package	TYPO3
 * @subpackage	tx_tweetbutton
 */
class tx_tweetbutton_pi1 extends tslib_pibase {
	var $prefixId      = 'tx_tweetbutton_pi1';		// Same as class name
	var $scriptRelPath = 'pi1/class.tx_tweetbutton_pi1.php';	// Path to this script relative to the extension dir.
	var $extKey        = 'tweetbutton';	// The extension key.
	var $pi_checkCHash = true;
	
	/**
	 * The main method of the PlugIn
	 *
	 * @param	string		$content: The PlugIn content
	 * @param	array		$conf: The PlugIn configuration
	 * @return	The content that is displayed on the website
	 */
	function main($content, $conf) {
		$this->conf = $conf;
		$this->pi_setPiVarDefaults();
		$this->pi_loadLL();

                // Check environment
                if (!isset($conf['data-count'])) {
                    return $this->pi_wrapInBaseClass($this->pi_getLL('no_ts_template'));
                }

                // Initialize the plugin configuration
                $this->init();

                // return html
                return $this->pi_wrapInBaseClass($this->showButton());
		
	}

        /**
     * Initializes the plugin configuration
     *
     */
    protected function init() {

        $this->pi_initPIflexForm();

        // Get values
        $this->conf['templateFile'] = (string) $this->fetchConfigurationValue('templateFile');
        $this->conf['data-count'] = (string) $this->fetchConfigurationValue('data-count');
        $this->conf['data-related'] = (string) $this->fetchConfigurationValue('data-related');
        $this->conf['data-via'] = (string) $this->fetchConfigurationValue('data-via');
        $this->conf['data-text'] = (string) $this->fetchConfigurationValue('data-text');
        $this->conf['data-url'] = (string) $this->fetchConfigurationValue('data-url');
        $this->conf['data-lang'] = (string) $this->fetchConfigurationValue('data-lang');

        // Set default values if necessary
        if (!$this->conf['templateFile']) {
            $this->conf['templateFile'] = 'EXT:' . $this->extKey . '/res/template.html';
        }
        if (!$this->conf['data-count']) {
            $this->conf['data-count'] = 'horizontal';
        }
        if (!$this->conf['data-url']) {
            $this->conf['data-url'] = 'http://' . $_ENV['SERVER_NAME'] . $_ENV['REQUEST_URI'];
        }
        if (!$this->conf['data-text']) {
            $this->conf['data-text'] = $GLOBALS['TSFE']->page['title'];
        }

        // Load template code
        $this->templateCode = $this->cObj->fileResource($this->conf['templateFile']);
    }

    /**
     * Fetches configuratin value given its name.
     * Merges flexform and TS configuration values.
     *
     * @param	string		$param Configuration value name
     * @return	string		HTML
     */
    protected function fetchConfigurationValue($param) {
        $value = trim($this->pi_getFFvalue($this->cObj->data['pi_flexform'], $param));
        return $value ? $value : $this->conf[$param];
    }

    /**
     * Generates the HTML output for the frontend
     *
     * @return	string		HTML output for FE
     */
    protected function showButton() {

        // Set mandatory parameters
        $markers['###DATA###'] .= ' data-count="' . $this->conf['data-count'] . '"';


        // Set optional parameters
        if ($this->conf['data-via']) {
            $markers['###DATA###'] .= ' data-via="' . $this->conf['data-via'] . '"';
        }
        if ($this->conf['data-text']) {
            $markers['###DATA###'] .= ' data-text="' . $this->conf['data-text'] . '"';
        }
        if ($this->conf['data-url']) {
            $markers['###DATA###'] .= ' data-url="' . $this->conf['data-url'] . '"';
        }
        if ($this->conf['data-lang']) {
            $markers['###DATA###'] .= ' data-lang="' . $this->conf['data-lang'] . '"';
        }
        // Set related accounts info
        if ($this->conf['data-related']){
            $markers['###DATA###'] .= ' data-related="' . $this->conf['data-related'] . '"';
        }


        // Get template for the button
        $template = $this->cObj->getSubpart($this->templateCode, '###BUTTON###');

        // Render output
        $content = $this->cObj->substituteMarkerArray($template, $markers);

        // Return HTML
        return $content;
    }




}



if (defined('TYPO3_MODE') && $TYPO3_CONF_VARS[TYPO3_MODE]['XCLASS']['ext/tweetbutton/pi1/class.tx_tweetbutton_pi1.php'])	{
	include_once($TYPO3_CONF_VARS[TYPO3_MODE]['XCLASS']['ext/tweetbutton/pi1/class.tx_tweetbutton_pi1.php']);
}

?>

pi1/class.tx_tweetbutton_pi1_wizicon.php
<?php
/***************************************************************
*  Copyright notice
*
*  (c) 2010 Peter Proell <peter@alinbu.net>
*  All rights reserved
*
*  This script is part of the TYPO3 project. The TYPO3 project is
*  free software; you can redistribute it and/or modify
*  it under the terms of the GNU General Public License as published by
*  the Free Software Foundation; either version 2 of the License, or
*  (at your option) any later version.
*
*  The GNU General Public License can be found at
*  http://www.gnu.org/copyleft/gpl.html.
*
*  This script is distributed in the hope that it will be useful,
*  but WITHOUT ANY WARRANTY; without even the implied warranty of
*  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
*  GNU General Public License for more details.
*
*  This copyright notice MUST APPEAR in all copies of the script!
***************************************************************/
/**
 * [CLASS/FUNCTION INDEX of SCRIPT]
 *
 * Hint: use extdeveval to insert/update function index above.
 */




/**
 * Class that adds the wizard icon.
 *
 * @author	Peter Proell <peter@alinbu.net>
 * @package	TYPO3
 * @subpackage	tx_tweetbutton
 */
class tx_tweetbutton_pi1_wizicon {

					/**
					 * Processing the wizard items array
					 *
					 * @param	array		$wizardItems: The wizard items
					 * @return	Modified array with wizard items
					 */
					function proc($wizardItems)	{
						global $LANG;

						$LL = $this->includeLocalLang();

						$wizardItems['plugins_tx_tweetbutton_pi1'] = array(
							'icon'=>t3lib_extMgm::extRelPath('tweetbutton').'pi1/ce_wiz.gif',
							'title'=>$LANG->getLLL('pi1_title',$LL),
							'description'=>$LANG->getLLL('pi1_plus_wiz_description',$LL),
							'params'=>'&defVals[tt_content][CType]=list&defVals[tt_content][list_type]=tweetbutton_pi1'
						);

						return $wizardItems;
					}

					/**
					 * Reads the [extDir]/locallang.xml and returns the $LOCAL_LANG array found in that file.
					 *
					 * @return	The array with language labels
					 */
					function includeLocalLang()	{
						$llFile = t3lib_extMgm::extPath('tweetbutton').'locallang.xml';
						$LOCAL_LANG = t3lib_div::readLLXMLfile($llFile, $GLOBALS['LANG']->lang);

						return $LOCAL_LANG;
					}
				}



if (defined('TYPO3_MODE') && $TYPO3_CONF_VARS[TYPO3_MODE]['XCLASS']['ext/tweetbutton/pi1/class.tx_tweetbutton_pi1_wizicon.php'])	{
	include_once($TYPO3_CONF_VARS[TYPO3_MODE]['XCLASS']['ext/tweetbutton/pi1/class.tx_tweetbutton_pi1_wizicon.php']);
}

?>

ext_emconf.php
<?php

########################################################################
# Extension Manager/Repository config file for ext "tweetbutton".
#
# Auto generated 13-08-2010 15:07
#
# Manual updates:
# Only the data in the array - everything else is removed by next
# writing. "version" and "dependencies" must not be touched!
########################################################################

$EM_CONF[$_EXTKEY] = array(
	'title' => 'The official twitter tweet button for your page',
	'description' => 'Put the official twitter tweet button on single pages or all pages',
	'category' => 'plugin',
	'author' => 'Peter Proell',
	'author_email' => 'peter@alinbu.net',
	'shy' => '',
	'dependencies' => '',
	'conflicts' => '',
	'priority' => '',
	'module' => '',
	'state' => 'stable',
	'internal' => '',
	'uploadfolder' => 0,
	'createDirs' => '',
	'modify_tables' => '',
	'clearCacheOnLoad' => 0,
	'lockType' => '',
	'author_company' => 'Alinbu.net TYPO3 Freelancers --- http://alinbu.net',
	'version' => '1.0.0',
	'constraints' => array(
		'depends' => array(
		),
		'conflicts' => array(
		),
		'suggests' => array(
		),
	),
	'_md5_values_when_last_written' => 'a:17:{s:9:"ChangeLog";s:4:"7a7c";s:10:"README.txt";s:4:"0194";s:12:"ext_icon.gif";s:4:"ada7";s:17:"ext_localconf.php";s:4:"42ac";s:14:"ext_tables.php";s:4:"4cb0";s:13:"locallang.xml";s:4:"8dd7";s:16:"locallang_db.xml";s:4:"1eb4";s:14:"doc/manual.sxw";s:4:"d634";s:14:"pi1/ce_wiz.gif";s:4:"1fa0";s:32:"pi1/class.tx_tweetbutton_pi1.php";s:4:"cdce";s:40:"pi1/class.tx_tweetbutton_pi1_wizicon.php";s:4:"cadb";s:13:"pi1/clear.gif";s:4:"cc11";s:19:"pi1/flexform_ds.xml";s:4:"f798";s:17:"pi1/locallang.xml";s:4:"27b9";s:17:"res/template.html";s:4:"aeb6";s:32:"static/tweetbutton/constants.txt";s:4:"dbf0";s:28:"static/tweetbutton/setup.txt";s:4:"6412";}',
	'suggests' => array(
	),
);

?>